### PR TITLE
Adding information about evaluating quantized models with GPUs

### PR DESCRIPTION
**Pull request by veekaybee**
## What's changing

When you run Ray evaluation job on a quantized model, it requires GPUs, per the bitsandbytes [quantization environment validation checker](https://github.com/huggingface/transformers/blob/fc37f38915372c15992b540dfcbbe00a916d4fc6/src/transformers/quantizers/quantizer_bnb_4bit.py#L74). 

We can make this clearer in the documentation for example evaluation runs. 

## How to test it

Check documenation

## Related Jira Ticket

Addressing this issue: https://github.com/mozilla-ai/lm-buddy/issues/50

## Additional notes for reviewers

Also added a config file for a quantized HF model eval run.

### CODE REVIEW

File: examples/notebooks/dev_ray_submission.ipynb
Comment (veekaybee) [inline examples/notebooks/dev_ray_submission.ipynb:1]: I wanted to make clearer where the path for the actual configs was relative to the notebooks, lmk if this is helpful.

File: README.md [hunk 1]
```diff
@@ -67,6 +67,7 @@ within a local Python driver script.
 This requires you to specify a Ray runtime environment containing:
 1) A `working_dir` for the local directory containing your job config YAML file, and
 2) A `pip` dependency for your desired version of `lm-buddy`.
+3) A `entrypoint_num_gpus` [parameter](https://docs.ray.io/en/latest/cluster/running-applications/job-submission/sdk.html#specifying-cpu-and-gpu-resources) if you are evaluating quantized models
```
Comment (sfriedowitz) [inline README.md:70]: This argument is not part of the `runtime_env` so worded separately.

```suggestion

Additionally, if your job requires GPU resources on the Ray entrypoint worker (
e.g., for loading large/quantized models), you should specify the
[entrypoint_num_gpus]https://docs.ray.io/en/latest/cluster/running-applications/job-submission/sdk.html#specifying-cpu-and-gpu-resources) parameter upon submission.
```

File: README.md [hunk 2]
```diff
@@ -79,12 +80,15 @@ client = JobSubmissionClient("http://127.0.0.1:8265")
 runtime_env = {
     "working_dir": "/path/to/working/directory",
     "pip": ["lm-buddy==X.X.X"]
+     # if you need GPUs for your workload
```
Comment (sfriedowitz) [inline README.md:83]:
```suggestion
```

not part of the runtime env

File: examples/notebooks/dev_ray_submission.ipynb
Comment (sfriedowitz) [reply examples/notebooks/dev_ray_submission.ipynb:1]: As it stands, this still cannot run.

If anything, you should specify the `working_dir` in the `runtime_env` to the `configs/simple` folder, and then in the entrypoint argument just do `simple_config.yaml`.

File: examples/notebooks/dev_ray_submission.ipynb [hunk 3]
```diff
@@ -110,11 +109,28 @@
...
+    "    entrypoint=f\"python -m lm_buddy run simple --config evaluation/lm_harness_quantized_config.yaml\",\n",
+    "    runtime_env=runtime_env,\n",
+    "    entrypoint_num_gpus=1\n",
+    ")"
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "b2ab4f17defeb574"
+  },
```
Comment (sfriedowitz) [inline examples/notebooks/dev_ray_submission.ipynb:133]: I don't think this is helpful for the `dev_ray_submission` context.

if anything, we should add explicit example notebooks for each job type.

Comment (veekaybee) [reply examples/notebooks/dev_ray_submission.ipynb:133]: Looks like [Mistral just released](https://github.com/mistralai/cookbook) a cookbook, too

Comment (sfriedowitz) [reply examples/notebooks/dev_ray_submission.ipynb:133]: For instance, an example "evaluating Mistral 7b across multiple axis", where we go through finetuning Mistral 7B and evaluating it on lm-harness, ragas, and prometheus, and compare and contrast results.

Comment (veekaybee) [reply examples/notebooks/dev_ray_submission.ipynb:133]: Yeah, that would be great - let's add it to the agenda.

File: examples/notebooks/dev_ray_submission.ipynb [hunk 4]
```diff
@@ -64,7 +64,7 @@
    "source": [
     "# Create a submission client bound to a Ray cluster\n",
     "# If using a remote cluster, replace 127.0.0.1 with the head node's IP address.\n",
-    "client = JobSubmissionClient(f\"http://127.0.0.1:8265\")"
+    "client = JobSubmissionClient(f\"http://10.140.58.130:8265\")"
```
Comment (sfriedowitz) [inline examples/notebooks/dev_ray_submission.ipynb:67]: This should be localhost still yeah?

File: examples/notebooks/dev_ray_submission.ipynb [hunk 5]
```diff
@@ -79,7 +79,8 @@
     "# but this is a convenient means to access the local module path\n",
     "import lm_buddy\n",
     "\n",
-    "lm_buddy_module = Path(lm_buddy.__file__).parent"
+    "lm_buddy_module = Path(lm_buddy.__file__).parent\n",
+    "root_dir = Path(lm_buddy.__file__).parents[2]"
```
Comment (sfriedowitz) [inline examples/notebooks/dev_ray_submission.ipynb:83]: ➕

**Review by sfriedowitz — APPROVED** [review]
Thanks for the iterations 👍

**Pull request merged**